truffleruby/src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -1085,4 +1087,32 @@ protected ObjectIVarSetNode createObjectIVarSetNode() {
 
     }
 
+    @CoreMethod(names = "store_exception", onSingleton = true, needsBlock = true)
+    public abstract static class StoreExceptionNode extends YieldingCoreMethodNode {
+
+        @Specialization
+        public Object executeWithProtect(VirtualFrame frame, DynamicObject block) {
+            try {
+                yield(frame, block);
+                return nil();
+            } catch (Throwable e) {
+                return e;
+            }
+        }
+    }
+
+    @CoreMethod(names = "raise_exception", onSingleton = true, required = 1)
+    public abstract static class RaiseExceptionNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        public Object executeWithProtect(Throwable e) {
+            if (e instanceof Error) {
+                throw (Error) e;
+            } else if (e instanceof RuntimeException) {
+                throw (RuntimeException) e;
+            }
+            throw new RuntimeException(e);
+        }
```

Comment:
The dark magic normally involves doing terrible things with generics (or maybe that loophole has been fixes). or using sun.misc.Unsafe, and I'm not sure either of those are a good idea, at least not without wrapping them in a more palatable form (like `ExceptionOperations.throw()`).